@Mr-Technician  What about the error from chat below.  Do you think maybe that was you running the build on top of  another build?
```
2>C:\Users\rnielsen\source\repos\MudBlazor\src\MudBlazor\MudBlazor.csproj(76,5): error MSB3491: Could not write lines to file "./TScripts/combined/MudBlazor.js". The process cannot access the file 'C:\Users\rnielsen\source\repos\MudBlazor\src\MudBlazor\TScripts\combined\MudBlazor.js' because it is being used by another process.
```

--- a/.github/workflows/deploy-nuget.yml
+++ b/.github/workflows/deploy-nuget.yml
@@ -1,27 +1,24 @@
-name: Pack and Publish Nuget Package
+name: deploy-nuget
 
 on:
   workflow_dispatch:
     inputs:
         nuget-version:
-          description: 'Nuget version v[0-9]+.[0-9]+.[0-9]+*'
+          description: 'Tag to publish v[0-9]+.[0-9]+.[0-9]+*'
           required: true
           default: '' 
           type: string
+  push:
+    tags: 
+      - "v[0-9]+.[0-9]+.[0-9]+*"
 env:
   VERSION: 1.0.0
-
-defaults:
-  run:
-    working-directory: ./src/MudBlazor
-    
+  
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - name: Checkout repository
-      uses: actions/checkout@v3
-    - name: Set version variable from input
+    - name: Set tag from input
       if: ${{ github.event.inputs.nuget-version != '' }}
       env:
         TAG: ${{ github.event.inputs.nuget-version }}
@@ -31,6 +28,10 @@ jobs:
       env:
         TAG: ${{ github.ref_name }}
       run: echo "VERSION=${TAG#v}" >> $GITHUB_ENV
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        ref: 'refs/tags/v${{ env.VERSION }}'
     - name: Setup dotnet version
       uses: actions/setup-dotnet@v3
       with:
@@ -39,6 +40,13 @@ jobs:
           7.0.x
     - name: Pack nuget package
       run: dotnet pack -c Release --output nupkgs /p:PackageVersion=$VERSION
+      working-directory: ./src/MudBlazor
+    #- name: Upload Artifact (for testing before publish)
+    #  uses: actions/upload-artifact@v3
+    #  with:
+    #    name: nuget
+    #    path: ./src/MudBlazor/nupkgs/*.nupkg
+    #    retention-days: 1
     - name: Publish nuget package
-      if: (github.ref_type == 'tag' && startsWith(github.ref, 'refs/tags/v')) || startsWith(${{ github.event.inputs.nuget-version }}, 'v')
       run: dotnet nuget push nupkgs/*.nupkg -k ${{ secrets.NUGET_KEY }} -s https://api.nuget.org/v3/index.json
+      working-directory: ./src/MudBlazor

--- a/src/MudBlazor.Docs/MudBlazor.Docs.csproj
+++ b/src/MudBlazor.Docs/MudBlazor.Docs.csproj
@@ -2,14 +2,6 @@
 <!--Use: dotnet msbuild -preprocess:<fileName>.xml to evaluate this project-->
 <Project Sdk="Microsoft.NET.Sdk.Razor">
 
-  <!--Hook for the generated static content-->
-  <PropertyGroup>
-    <ResolveStaticWebAssetsInputsDependsOn>
-      IncludeGeneratedStaticFiles;
-      $(ResolveStaticWebAssetsInputsDependsOn)
-    </ResolveStaticWebAssetsInputsDependsOn>
-  </PropertyGroup>
-
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
   </PropertyGroup>
@@ -112,13 +104,14 @@
   </Target>
 
   <!--Output of Excubo webcompiler-->
-  <Target Name="IncludeGeneratedStaticFiles" DependsOnTargets="WebCompiler">
-    <Error Condition="!Exists('wwwroot/MudBlazorDocs.min.css')" Text="Missing MudBlazorDocs.min.css in wwwroot" />
+  <Target Name="IncludeGeneratedStaticFiles" DependsOnTargets="WebCompiler" BeforeTargets="BeforeBuild">
+    <Error Condition="!Exists('$(MSBuildProjectDirectory)/wwwroot/MudBlazorDocs.min.css')" Text="Missing MudBlazorDocs.min.css in wwwroot" />
     <!--If we ever bundle js  for the docs site uncomment the below check-->
     <!--<Error Condition="!Exists('wwwroot/MudBlazorDocs.min.js')" Text="Missing MudBlazorDocs.min.js in wwwroot" />-->
     <ItemGroup>
-      <Content Include="wwwroot/MudBlazorDocs.min.css" Condition="!Exists('wwwroot/MudBlazorDocs.min.css')" />
-      <Content Include="wwwroot/MudBlazorDocs.min.js" Condition="!Exists('wwwroot/MudBlazorDocs.min.js')" />
+      <!--Include without duplication-->
+      <_NewCompiledDocsCssFiles Include="wwwroot\MudBlazorDocs.min.css" Exclude="@(Content)" />
+      <Content Include="@(_NewCompiledDocsCssFiles)" />
     </ItemGroup>
   </Target>
 
@@ -126,9 +119,6 @@
   <ItemGroup>
     <Content Remove="compilerconfig.json" />
     <Content Remove="excubowebcompiler.json" />
-  </ItemGroup>
-  <ItemGroup>
-    <Folder Include="wwwroot\images\landingpage" />
   </ItemGroup>
 
 </Project>

--- a/src/MudBlazor/MudBlazor.csproj
+++ b/src/MudBlazor/MudBlazor.csproj
@@ -1,13 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Razor">
 
   <PropertyGroup>
-    <ResolveStaticWebAssetsInputsDependsOn>
-      IncludeGeneratedStaticFiles;
-      $(ResolveStaticWebAssetsInputsDependsOn)
-    </ResolveStaticWebAssetsInputsDependsOn>
-  </PropertyGroup>
-
-  <PropertyGroup>
     <TargetFrameworks>net6.0;net7.0</TargetFrameworks>
   </PropertyGroup>
 
@@ -90,12 +83,15 @@
     <Exec Command="dotnet webcompiler ./TScripts/combined/MudBlazor.js -c excubowebcompiler.json" StandardOutputImportance="high" StandardErrorImportance="high" />
   </Target>
 
-  <Target Name="IncludeGeneratedStaticFiles" DependsOnTargets="WebCompiler">
-    <Error Condition="!Exists('wwwroot/MudBlazor.min.css')" Text="Missing MudBlazor.min.css in wwwroot" />
-    <Error Condition="!Exists('wwwroot/MudBlazor.min.js')" Text="Missing MudBlazor.min.js in wwwroot" />
+  <Target Name="IncludeGeneratedStaticFiles" DependsOnTargets="WebCompiler" BeforeTargets="BeforeBuild">
+    <Error Condition="!Exists('$(MSBuildProjectDirectory)/wwwroot/MudBlazor.min.css')" Text="Missing MudBlazor.min.css in wwwroot" />
+    <Error Condition="!Exists('$(MSBuildProjectDirectory)/wwwroot/MudBlazor.min.js')" Text="Missing MudBlazor.min.js in wwwroot" />
     <ItemGroup>
-      <Content Include="wwwroot/MudBlazor.min.css" Condition="!Exists('wwwroot/MudBlazor.min.css')" />
-      <Content Include="wwwroot/MudBlazor.min.js" Condition="!Exists('wwwroot/MudBlazor.min.js')" />
+      <!--Include without duplication-->
+      <_NewCompiledCssFiles Include="wwwroot\MudBlazor.min.css" Exclude="@(Content)" />
+      <_NewCompiledJsFiles Include="wwwroot\MudBlazor.min.js" Exclude="@(Content)" />
+      <Content Include="@(_NewCompiledCssFiles)" />
+      <Content Include="@(_NewCompiledJsFiles)" />
     </ItemGroup>
   </Target>
 
@@ -103,11 +99,9 @@
   <ItemGroup>
     <Content Remove="compilerconfig.json" />
     <Content Remove="excubowebcompiler.json" />
-    <Content Remove="**/*/DoNotRemove.txt" />
-  </ItemGroup>
-  <ItemGroup>
-    <Folder Include="TScripts\combined\" />
-    <Folder Include="wwwroot\" />
+    <Content Remove="wwwroot/DoNotRemove.txt" />
+    <!--macOS hidden file (causes problems with dotnet pack)-->
+    <Content Remove="**/*/.DS_Store" />
   </ItemGroup>
 
   <ItemGroup>

